src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/DoNotAddFinilizerToMemoryManagerTDerivedTypesTests.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Testing;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Runtime.DoNotAddFinalizerToMemoryManagerTDerivedTypes,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+
+namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
+{
+    public class DoNotAddFinilizerToMemoryManagerTDerivedTypesTests
+    {
+        [Fact]
+        public async Task ClassNotDerivedFromMemoryManagerOK()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+using System;
+
+namespace TestNamespace
+{
+    class TestClass
+    {
+        private void TestMethod(string test)
+        {
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task ClassDerivedFromMemoryManagerNoFinilizerOK()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+using System;
+using System.Buffers;
+
+namespace TestNamespace
+{
+    abstract class TestClass<T> : MemoryManager<T>
+    {
+    }
+}");
+        }
+
+
+        [Fact]
+        public async Task ClassDerivedFromMemoryManagerWithFinilizerWarns()
```

Comment:
You should also add a test like

```C#
class Deeper : Middle
{
    ~Deeper() { }
}

class Middle : MemoryManager<byte>
{
    ...
}
```

And tests where Middle and Deeper are both abstract.